api/src/main/java/org/openmrs/module/idgen/IdentifierSource.java

```diff
@@ -170,4 +170,34 @@
 	 * @param - the reserved identifier to add
 	 */
 	public abstract void addReservedIdentifier(String reservedIdentifier);
+        
+        /**
+	 * @return Set of auto generation options
+	 */
+	public abstract Set<String> getAutoGenerationOptions();
```

Comment:
What is the reason behind adding these new methods?

Reply:
When I add a new `<set />`, I need to add the setters and getters in IdentifierSource.java class otherwise I get the following errors:

`Caused by: org.hibernate.PropertyNotFoundException: Could not find a getter for autoGenerationOptions in class org.openmrs.module.idgen.IdentifierSource`

The sets are needed so that i can declare the `cascade="all-delete-orphan"` to delete the children in related tables.

Reply:
Remove them and i see the error.

Reply:
removed